src/Transport/Receiving/MessagePump.cs

```diff
@@ -129,7 +129,6 @@ public void Start(PushRuntimeSettings limitations)
                 // TODO: remove workaround when https://github.com/Azure/azure-service-bus-dotnet/issues/439 is fixed
                 Interlocked.Increment(ref numberOfExecutingReceives);
                 message = await receiveTask.ConfigureAwait(false);
-                Interlocked.Decrement(ref numberOfExecutingReceives);
```

Comment:
If receive task throws the counter would not be decremented

Reply:
Such a horrible implementation just because ASB client doesn't take a cancellation token into receive task, sigh.